### PR TITLE
fix(generator): escape special chars for path replacements with regex

### LINKED ISSUE #32: Wrong output location, if template path contains special characters
If the project path contains special characters, the code for creating the `target_file` path in [generator.js (line 100 and following)](https://github.com/fmvilas/openapi3-generator/blob/master/lib/generator.js#L100) doesn't work as intended.

e.g. if the paths look like this:
```
template path:  /some+dir/project/templates/some-file
templates root: /some+dir/project/templates
```
the replacement of `/some+dir/project/templates` in the template file path doesn't work, because the `+` is interpreted as a regular expression instruction.

I think all special chars for regular expressions should be escaped here.

### PR DESCRIPTION
**Pull request by richardklose**
If a projects path contains special characters, they will be intrepeted as
regular expresion instructions, which results in wrong output paths in the
end. To avoid this, special characters that can be used in regular
expressions should be escaped.

Fixes #32

### CODE REVIEW

**Comment by fmvilas** [comment]
Hey Richard! I'm not maintaining this project anymore. I love it but I don't have time. You're probably the person who did the most here aside from me so I'm happy to lend it to you if you're in the mood to maintain it.

Related: https://twitter.com/fmvilas/status/1167504577228353537

**Comment by richardklose** [comment]
Oh, that's sad to hear! You did a great job with this project so far. I personally don't have the time to maintain this project alone, but I could do so within my work at Auerswald. So If this is an option for you, we can maintain it as a company. We are already using openapi3-generator in some of our products and plan to use it in more, so as long as we continue work on our products for the next years, we will keep an eye on this.

**Comment by richardklose** [comment]
@philsturgeon as it looks that you are the new maintainer now, I'd like to ask, if this can be merged and released or if there is anything that should be added, before this can be merged?

**Review by philsturgeon — APPROVED** [review]

**Comment by philsturgeon** [comment]
@richardklose im the host currently but not actively maintaining it. Would you be up to that challenge?

**Comment by fmvilas** [comment]
@richardklose I think you should be the one man! 🙌

**Pull request merged**
